xmppserver/src/main/java/org/jivesoftware/openfire/net/SocketReader.java

```diff
@@ -461,8 +463,12 @@ private boolean isHostUnknown(String host) {
             // requested host matched the server name
             return false;
         }
-        // Check if the host matches a subdomain of this host
-        return !routingTable.hasComponentRoute(new JID(host));
+        if (routingTable.hasComponentRoute(new JID(host))) {
+            // Check if the host matches a subdomain of this host
+            return false;
+        }
+
+        return !JiveGlobals.getListProperty("xmpp.gateway.domains", Collections.emptyList()).contains(host);
```

Comment:
Why JiveGlobals rather than a SystemProperty?

Reply:
Lazy. I will change this.